### PR TITLE
Readme draft

### PR DESCRIPTION
**Pull request by yury-gorbachev**
Updates to describe:
- latest changes in library with focus on GenAI api
- expose more functionality that we have (continuous batching, speculative decoding, etc.)
- provide lightweight samples as a beginning

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,51 +1,238 @@
...
+- Select active adapters for every generation
+- Mix multiple adapters with coefficients via alpha blending
+
+All scenarios are run on top of OpenVINO Runtime that supports inference on CPU, GPU and NPU. See [here](https://docs.openvino.ai/2024/about-openvino/release-notes-openvino/system-requirements.html) for platform support matrix.
+
+## Supported Generative AI optimization methods
+
+OpenVINO™ GenAI library provides transparent way to use state of the art generation optimizations:
+- Speculative decoding that employs two models of different size and uses large model to periodically correct results of small model. See [here](https://pytorch.org/blog/hitchhikers-guide-speculative-decoding/) for more detailed overview
+- Lookahead decoding that attempts to guess multiple tokens based on historical context and corrects this guess using LLMs. See [here](https://lmsys.org/blog/2023-11-21-lookahead-decoding/) for more detailed overview
```
Comment (ilya-lavrenov) [inline README.md:28]: `Lookahead decoding` is not supported

File: README.md [hunk 2]
```diff
@@ -1,51 +1,238 @@
...
+- KVCache token eviction algorithm that reduces size of the KVCache by pruning less impacting tokens.
+
+Additionally, OpenVINO™ GenAI library implements continuous batching approach to use OpenVINO within LLM serving. Continuous batching library could be used in LLM serving frameworks and supports following features:
```
Comment (ilya-lavrenov) [inline README.md:31]: should we also refer to OVMS docs as entry point to Continuous batching functionality?

File: README.md [hunk 3]
```diff
@@ -1,51 +1,238 @@
...
+```sh
+    # Installing OpenVINO GenAI via pip
+    pip install openvino-genai
+
+    # Install optimum-intel to be able to download, convert and optimize LLMs from Hugging Face
+    # Optimum is not required to run models, only to convert and compress
+    pip install optimum[openvino]
+
+    # (Optional) Install (TBD) to be able to download models from Model Scope
+    #pip install optimum[openvino]
```
Comment (ilya-lavrenov) [inline README.md:45]: see https://pypi.org/project/modelscope/

Looks like we can recommend `pip install modelscope` with additional extras for NLP and other domains

File: README.md [hunk 4]
```diff
@@ -1,51 +1,238 @@
...
+```
+
+## Performing text generation 
+<details>
+For more examples check out our [LLM Cheat Sheet](https://docs.openvino.ai)
```
Comment (ilya-lavrenov) [inline README.md:50]: should it point to https://docs.openvino.ai/2024/learn-openvino/llm_inference_guide.html ?

File: samples/generation.gif
Comment (ilya-lavrenov) [inline samples/generation.gif:1]: need to put such images under LFS similarly to https://github.com/openvinotoolkit/openvino/blob/master/.gitattributes#L64

File: README.md [hunk 5]
```diff
@@ -1,51 +1,238 @@
...
+   std::cout << pipe.generate("The Sun is yellow because", ov::genai::max_new_tokens(100));
+}
+```
+
+### Sample notebooks using this API
+
+(TBD)
+
+</details>
+
```
Comment (ilya-lavrenov) [inline README.md:91]: should we also point out to a list of supported models? https://github.com/openvinotoolkit/openvino.genai/blob/master/src/docs/SUPPORTED_MODELS.md#large-language-models

File: README.md (hunk 3 above)
Comment (yury-gorbachev) [reply README.md:45]: This is good thing to have, but I am not sure if conversion will work as it is. 
Proposal is to make this fix later when we will have document on modelscope support itself (this is being worked on right now)

File: README.md (hunk 4 above)
Comment (yury-gorbachev) [reply README.md:50]: Yes, I will use this now, but original intent was to make cheat sheet for GenAI specifically. 
We can update readme later.

**Review by ilya-lavrenov — APPROVED** [review]

**Pull request merged**